plugins/arElasticSearchPlugin/lib/arElasticSearchPlugin.class.php

```diff
@@ -368,6 +368,31 @@ public function partialUpdate($object, $data)
         }
     }
 
+    public function partialUpdateById(string $className, int $id, array $data)
+    {
+        if (!$this->enabled) {
+            return;
+        }
+
+        if (0 == strcmp($className, 'QubitUser')) {
+            return;
+        }
+
+        $document = new \Elastica\Document($id, $data);
+
+        try {
+            $this->index->getType($className)->updateDocument($document);
+        } catch (\Elastica\Exception\ResponseException $e) {
+            // Create document if no existing document exists
```

Comment:
Not my first language but that feels like a lot of wording :P

Reply:
True, "if none exists" would suffice.